design-documents/provisioning/configuration.md

```diff
@@ -64,26 +81,174 @@ The usual behavior for installing from scratch or upgrading is the following:
 * Recurring Data
 * app:config:import
 
-The configuration provisioning core logic should be executed as one of the last recurring data scripts. As alternative the execution sequence mentioned above can be changed so that it runs directly after recurring data and before config import.
+There is a dependency tree during installation like the following:
+
+* Declarative Schema 
+* Schema Patches 
+* Recurring Schema Scripts
+* Websites => Store Groups => Stores 
+* System Configuration 
+* Attributes => Attribute Sets / Attribute Groups
+* Tax 
+* Data Patches
+* Recurring Data Scripts
+* app:config:import
 
-The reason for this is that also registered themes (inside a recurring data script) can still be used by the provisioning logic. This would be obsolete if theme registration is part of configuration provisioning.
 
 #### Data structure
 
 As already mentioned the data structure is related to the existing database declarative schema. Every module can define such a XML file for a configuration type. All configuration files for the same type would be merged to one big logical structure. Therefore every configuration type module should at least support merge logic. This is done individually as for example websites are merged differently than attributes.
 
 Therefore a virtual type of `Magento\Framework\Config\Reader\Filesystem` will be defined which has a ` Magento\Framework\Config\ConverterInterface` which is responsible for merging the XML files.
 
-On top of that every configuration type module defines a `maintained` flag which basically defines if the configuration can be overriden by a user or not. If it is set to `yes` the configuration provisioning will always override the values, independently if it was changed or not.
+There is also a field `is_mutable` which defines if the declared data can be changed by the admin. The following explains the behavior:
+
+* `is_mutable` is not defined:
+  * new entity is installed
+  * checkbox is unchecked
+  * admin can change the entity (after checking the checkbox)
+  * won't overwrite admin side manual created entities until checkbox is unchecked
+* `is_mutable = false`:
+  * new entity is installed
+  * checkbox does not exist
+  * admin can't change the entity
+  * will overwrite admin side manual created entities
+
+#### Example: Attribute behavior
+ - "**is_mutable**" flag is checkbox at the attribute grid and at attribute edit page
+ - Attributes which created from Admin panel is not maintainable by xml (no dumping, "**is_mutable**" flag is not shown but has true value in database) until xml with the same attribute code will be created and "**is_mutable**" will be unchecked
+ - "**is_mutable**" flag is responsible for defining maintenance responsibilities:
+   - **true** responsibility on Admin panel side switchable from admin
```

Comment:
Why `true` and `false` options are needed? Why would we need for admin to check/uncheck the checkbox?

Reply:
I'd reiterate on description for "is_mutable". It looks a little bit confusing right now, from different sentences.
Would it even make sense to divide it into two options?

1. `is_mutable` (`true`, `false`) - if true, admin can modify the value (which will override what's declared in the file), if false, admin can't modify it
2. revert to system value - a button/checkbox for mutable attributes in Admin UI to revert to system value (declared in the file)

Case of removal should be also covered. Should it be possible for admin to remove a mutable entity/attribute? In this case, should there be UI for restoring it from the system value?